core/src/main/scala/org/apache/spark/deploy/rest/StandaloneRestServer.scala

```diff
@@ -138,6 +138,10 @@ private[rest] class StandaloneSubmitRequestServlet(
     val driverExtraClassPath = sparkProperties.get("spark.driver.extraClassPath")
     val driverExtraLibraryPath = sparkProperties.get("spark.driver.extraLibraryPath")
     val superviseDriver = sparkProperties.get("spark.driver.supervise")
+    val masters = sparkProperties.get("spark.master")
+    val masterPort = Utils.extractHostPortFromSparkUrl(masterUrl)._2.toString
```

Comment:
port should be an int here, really, right? elsewhere it might be used as a string but that's separate.
You can write `val (_, masterPort) = Utils....`